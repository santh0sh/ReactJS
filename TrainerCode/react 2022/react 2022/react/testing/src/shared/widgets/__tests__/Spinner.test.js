import { mount } from 'enzyme';
import Spinner from '../Spinner';

describe('Render Spinner', () => {
    it('render correctly Spinner component', () => {
        const SpinnerComponent = mount(<Spinner />);
        expect(SpinnerComponent).toMatchSnapshot();
    });

    it('check prop title by default', () => {
        const SpinnerComponent = mount(<Spinner />);
        expect(SpinnerComponent.find('p').text()).toEqual('Please wait');
    });

    it('check prop type for title is string', () => {
        const props = {
                title: 'Wait'
            },
            SpinnerComponent = mount(<Spinner {...props} />);
        expect(SpinnerComponent.find('p').text()).toBeString();
    });

    it('check prop title with html tags', () => {
        const props = {
                title: '<b>Please wait</b>'
            },
            SpinnerComponent = mount(<Spinner {...props} />);
        expect(SpinnerComponent.find('p').text()).toEqual('Please wait');
    });

    describe('check prop subTitle', () => {
        const props = {
                subTitle: 'left 1 minute'
            },
            SpinnerComponent = mount(<Spinner {...props} />);

        it('type for subTitle is string', () => {
            expect(SpinnerComponent.find('p').at(1).text()).toBeString();
        });

        it('render correct text', () => {
            expect(SpinnerComponent.find('p').at(1).text()).toEqual(props.subTitle);
        });
    });

    it('check subTitle is not rendered', () => {
        const SpinnerComponent = mount(<Spinner />);
        expect(SpinnerComponent.find('p').length).toEqual(1);
    });
});
