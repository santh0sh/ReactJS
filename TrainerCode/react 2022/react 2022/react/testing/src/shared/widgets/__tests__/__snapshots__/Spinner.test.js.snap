// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Spinner render correctly Spinner component 1`] = `
<_class>
  <Spinner
    title="Please wait"
  >
    <div
      className="spinner"
    >
      <div
        className="spinner-overlay"
      >
        Â 
      </div>
      <div
        className="spinner-content"
      >
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "Please wait",
            }
          }
        />
        <Loader
          loaderLineCount={8}
        >
          <div
            className="loader"
          >
            <span
              key="0"
            />
            <span
              key="1"
            />
            <span
              key="2"
            />
            <span
              key="3"
            />
            <span
              key="4"
            />
            <span
              key="5"
            />
            <span
              key="6"
            />
            <span
              key="7"
            />
          </div>
        </Loader>
      </div>
    </div>
  </Spinner>
</_class>
`;
